specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json

```diff
@@ -994,6 +1034,48 @@
         "tableName"
       ]
     },
+    "RestServiceDataset": {
```

Comment:
By convention the model name is discriminator value + dataset, so should be RestResourceDataset? Any reason it is different?

Reply:
@dazhao-msft, please comment.

Reply:
Some background information to share: This part is referenced from the private swagger which is already used by UI.

Reply:
Davzhao will help comment.

Reply:
As long as there are no backend changes which could cause a break for customers we should make our model consistent.

Reply:
Hvermis, if we change the name here, is there any issue before the change of private swagger is done?

Reply:
There shouldn't be. You will need to update the UI code to support both names for some time, then update the private swagger, then cleanup the UI code.

Reply:
UI code doesn't use this name.

Reply:
You should do the changes to the swagger, generate the .Net SDK, and test out these types. Add serialization tests, and then use the jsons generated by SDK and test agains REST endpoint. This will make sure that your changes will work in SDK.